### ISSUE TITLE
forge-create-issue produces a confusing error when repo has no clone

### ISSUE THREAD

**Issue by bcc32** [magit/forge#722]
When attempting to create an issue for a repo that has been added to
forge but which has no local clone, the error produced is confusing:

    Creating directory: Permission denied, /magit

It seems to be attempting to create a directory to store the draft
issue text, which makes sense, but the error itself is a little
unclear and it would probably be a nicer experience to handle this
case specifically and produce a `user-error`.

Steps to reproduce are just:

1. M-x forge-add-repository and specify some GitHub repo you don't
   have a clone for.  I used djcb/mu.
2. M-x forge-list-repositories and go to the topics page for that
   newly-added repo.
3. M-x forge-create-issue, which errors with the message mentioned
   above.

Backtrace follows:

```
Debugger entered--Lisp error: (permission-denied "Creating directory" "Permission denied" "/magit")
  transient--exit-and-debug(error (permission-denied "Creating directory" "Permission denied" "/magit"))
  signal(permission-denied ("Creating directory" "Permission denied" "/magit"))
  files--ensure-directory(make-directory-internal "/magit")
  make-directory("/magit/posts/" t)
  forge--prepare-post-buffer("new-issue" "Create new issue on djcb/mu")
  #<subr forge-create-issue>()
  apply(#<subr forge-create-issue> nil)
  #f(compiled-function (fn &rest args) #<bytecode 0x10411af0915d810c>)(#<subr forge-create-issue>)
  apply(#f(compiled-function (fn &rest args) #<bytecode 0x10411af0915d810c>) #<subr forge-create-issue> nil)
  (lambda (fn &rest args) (interactive #f(compiled-function (spec) #<bytecode 0x43d85eb2db5920d>)) (apply '#f(compiled-function (fn &rest args) #<bytecode 0x10411af0915d810c>) fn args))(#<subr forge-create-issue>)
  apply((lambda (fn &rest args) (interactive #f(compiled-function (spec) #<bytecode 0x43d85eb2db5920d>)) (apply '#f(compiled-function (fn &rest args) #<bytecode 0x10411af0915d810c>) fn args)) #<subr forge-create-issue> nil)
  forge-create-issue()
  funcall-interactively(forge-create-issue)
  command-execute(forge-create-issue)
```

FYI version info:

    M-x magit-version RET

Magit 20241020.1608 [>= 20241020.1608], Transient 20241018.1740, Forge 20241018.1648, Git 2.44.1, Emacs 29.4, gnu/linux

**Comment by tarsius** [comment]
Yes, the errors could be improved and it is also worth looking into instead making it possible to create issues in such situations (and pull-requests, but that is both harder and stranger (where does the branch come from if there is no clone?)).

Additional work is arriving at a very high frequency right now, so I won't get to this soon.

**Comment by bcc32** [comment]
No problem, thanks for the ack and all your amazing work on magit and forge!

**Comment by tarsius** [comment]
It's now possible to create new issue and comment on existing topics without a local clone. New pull-requests still cannot be created in this case, but attempting to do so now results in a more informative error.

**Comment by bcc32** [comment]
Thank you!
